src/lints/enum_variant_field_deprecated.ron

```diff
@@ -0,0 +1,66 @@
+SemverQuery(
+    id: "enum_variant_field_deprecated",
+    human_readable_name: "enum variant field marked with #[deprecated]",
+    description: "A field in an enum variant was marked with the #[deprecated] attribute.",
+    required_update: Minor,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/reference/attributes/diagnostics.html#the-deprecated-attribute"),
+    query: r#"
+    {
+        CrateDiff {
+            current {
+                item {
+                    ... on Enum {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        enum_name: name @output
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+                        variant {
+                            ... on StructVariant {
```

Comment:
Tuple variants can also have deprecated fields:
```rust
pub enum Example {
    Variant(#[deprecated] i64),
}
```

We'll need to add separate lints for struct and tuple variants, so let's name this lint `enum_struct_variant_field_deprecated` and the tuple one `enum_tuple_variant_field_deprecated`.